File: sources/academy/webscraping/scraping_basics_python/exercises/imo_countries_single_selector.py
import httpx
from bs4 import BeautifulSoup

url = "https://www.imo.org/en/ourwork/ero/pages/memberstates.aspx"
response = httpx.get(url)
response.raise_for_status()

soup = BeautifulSoup(response.text, "html.parser")

for cell in soup.select(".content table tr td:nth-child(odd)"):
    if name := cell.text.strip():
        print(name)

File: sources/academy/webscraping/scraping_basics_javascript/exercises/imo_countries_single_selector.mjs
import * as cheerio from 'cheerio';

const url = 'https://www.imo.org/en/ourwork/ero/pages/memberstates.aspx';
const response = await fetch(url);

if (!response.ok) {
  throw new Error(`HTTP ${response.status}`);
}

const html = await response.text();
const $ = cheerio.load(html);

for (const element of $('.content table tr td:nth-child(odd)').toArray()) {
  const name = $(element).text().trim();
  if (name) {
    console.log(name);
  }
}
